Google.Api.Gax/VersionHeaderBuilder.cs

```diff
@@ -112,6 +112,35 @@ private static string FormatAssemblyVersion(System.Type type)
             return FormatVersion(assembly.GetName().Version);
         }
 
+        // Visible for testing
+
+        /// <summary>
+        /// Formats an AssemblyInformationalVersionAttribute value to avoid losing useful information,
+        /// but also avoid including unnecessary hex that is appended automatically.
+        /// </summary>
+        internal static string FormatInformationalVersion(string info)
+        {
+            // In some case, the runtime includes 40 hex digits after a + or period in InformationalVersion.
+            // In precisely those cases, we strip this.
+            int signIndex = Math.Max(info.LastIndexOf('.'), info.LastIndexOf('+'));
+            if (signIndex == -1 || signIndex != info.Length - 41)
+            {
+                return info;
+            }
+            for (int i = signIndex + 1; i < info.Length; i++)
+            {
+                char c = info[i];
+                bool isHex = (c >= '0' && c <= '9')
+                    || (c >= 'a' && c <= 'f')
+                    || (c >= 'A' && c <= 'F');
+                if (!isHex)
```

Comment:
Well, it could also all be hex characters and "valid" info. I guess what am saying is that the fact that `preview` and `build` use characters outside of hex is just a coincidence, there might be other similar qualifiers that use all hex characters.

If this is used by us alone, we know which qualifiers we use, and we can skip stripping those. Else we may be stripping things that someone intended to have as part of the version?

I know this is a nit-pick though, so it's fine as is.

Reply:
It would have to be exactly 40 characters, all hex digits, following a period or `+`. I've been deliberately as picky as I can be, basically :)

Reply:
Oh, didn't notice the length checking before the for. Yep that probably makes it way better. Thanks!